docs/source/examples/dataframe-hdf5.rst

```diff
@@ -0,0 +1,146 @@
+
+.. code-block:: python
+   :emphasize-lines: 40,45
+   
+    
+    def create_hdf5s(iters):
+           
+       import random
+       import pandas as pd
+       import numpy as np
+       import string 
+       import json
+    
+       # Intitalize counter to build specific number of hdf5 files
+       count = 0
+       setseed=1
+       np.random.seed(seed=setseed)
+       random.seed(setseed)
+    
+       # Key to store group keys and file names for generated hdf5 files
+       fileKeys = {}
+    
+       while count <= iters:
+            
+           # randomly pick letter as dataset key
+           groupkey = random.choice(list(string.ascii_lowercase))
+        
+           # randomly pick a number as hdf5 filename
+           saver = 'my'+str(np.random.randint(100))+'.h5'
+        
+           # Make a dataframe; 26 rows, 2 columns
+           df = pd.DataFrame(
+           {'x': np.random.randint(1,1000,26),
+           'y': np.random.randint(1,1000,26)},
+           index=list(string.ascii_lowercase)
+           )
+        
+           # Write synthetic hdf5 to current directory
+           df.to_hdf(saver, key='/'+groupkey, format='table')
+           fileKeys[saver]=groupkey
+           count += 1
+           setseed+=1
+        
+       # Return the dictionary with file key and filename for testing
+       with open('hdf5Keys.json','w') as f:
+           f.write(json.dumps(fileKeys))
+
+    return fileKeys
+
+    # Create the HDF5 files and keys to access
+    hdf5Keys = create_hdf5s(keys)
+
+
+.. doctest:: 
+    
+    def create_hdf5s(iters):
+        import random
+        import pandas as pd
+        import numpy as np
+        import string 
+        import json
+
+        # Intitalize counter to build specific number of hdf5 files
+        count = 0
+
+        # Key to store group keys and file names for generated hdf5 files
+        fileKeys = {}
+
+        while count <= iters:
+            # randomly pick letter as dataset key
+            groupkey = random.choice(list(string.ascii_lowercase))
+
+            # randomly pick a number as hdf5 filename
+            saver = 'my'+str(np.random.randint(100))+'.h5'
+
+            # Make a dataframe; 26 rows, 2 columns
+            df = pd.DataFrame(
+            {'x': np.random.randint(1,1000,26),
+            'y': np.random.randint(1,1000,26)},
+            index=list(string.ascii_lowercase)
+            )
+
+            # Write synthetic hdf5 to current directory
+            df.to_hdf(saver, key='/'+groupkey, format='table')
+
+            fileKeys[saver]=groupkey
+
+            count += 1
+        # Return the dictionary with file key and filename for testing
+        with open('hdf5Keys.json','w') as f:
+            f.write(json.dumps(hdf5Keys))
+        return fileKeys
+    
+    # Build the exercise data
+    hdf5Keys = create_hdf5s(keys)
+        
+**Read single dataset from HDF5:**
```

Comment:
Honestly, the pandas docs are pretty awful here -- we should probably update them! So I guess this is probably best handled upstream. 

I think the pytables FAQ covers this OK in the entry comparing it to h5py: https://github.com/dask/dask/pull/1413

The thing I would note for dask is that it supports the pandas flavor of HDF5 (which itself is a subtype of pytables supported HDF5). Fixed storage is just one of the types of HDF5 data pytables supports on top of HDF5.

This will probably not be surprising to any pandas users but users who didn't write their HDF5 files with pandas will almost certainly be unable to read them with dask.dataframe.